extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -231,12 +232,12 @@ public Long requestContentLengthUncompressed(final HttpRequest request, final Ht
 
         @Override
         public Integer statusCode(final HttpRequest request, final HttpResponse response) {
-            return response.statusCode();
+            return response != null ? response.statusCode() : 499;
         }
 
         @Override
         public Long responseContentLength(final HttpRequest request, final HttpResponse response) {
-            return getContentLength(response.headers());
+            return response != null ? getContentLength(response.headers()) : 0;
```

Comment:
Please, return `null` here if the `response` is null. The extractor can handle such return value.